sopel/irc/__init__.py

```diff
@@ -30,83 +30,91 @@
 import os
 import threading
 import time
-from typing import Optional
+from typing import Any, Callable, Dict, Iterable, List, Optional, Set, Tuple, TYPE_CHECKING
 
 from sopel import tools, trigger
 from sopel.tools import identifiers
```

Comment:
A change to this import from my initial review was applied in 596a58b23052f5acd2eea1253fe5268c6614758f but then reverted by 4597533fc0cde79c099923147d56a965b503e78d. @Exirel Careful when squashing this, because you probably don't want to try applying "fix 1c77c67" directly to that commit. x)

Reply:
Yup, revert because another part of the file is now using a bunch of features from `sopel.tools.identifiers`.